platforms/android/run_module_app/README.md

```diff
@@ -0,0 +1,47 @@
+# iree-run-module Android App
+
+This directory contains an Android application packing the `iree-run-module`
+command-line tool together with a specific IREE VM module invocation.
+This is helpful for profiling/debugging using Android tools.
+
+## Native Activity
+
+The App uses Android [`NativeActivity`][native-activity] to bridge IREE core
+libraries together with the Android system. Native activity allows one to
+implement an Android Activity purely in C/C++. There are
+[tutorials][native-activity-tutorial] and [examples][native-activity-example]
+one can follow to learn about Native Activity.
```

Comment:
We can do that, but I won't worry too much here given the purpose of this app is for development/profiling, instead of demonstrating what would be the expected integration path with IREE for a proper Android app. The later is super important too and we should certainly have that. (I want to see great integration with other Android functionalities with almost zero copy, sharing semaphores, etc. ;-P) For development, isolated binaries and reduced number of layers for simplicity is nice to have. :)

`build_apk.sh` is not there because of using `NativeActivity`. It's there to avoid using Android Studio, which can be cumbersome because we need to pack both the shared library and a concrete VM bytecode module and its invocation information together into an Android app. One can choose to use Android Studio and not use the script; just need to throw in a few gradle scripts, e.g., https://github.com/googlesamples/android-vulkan-tutorials/tree/master/tutorial01_load_vulkan. But managing the VM bytecode and invocation information will be cumbersome and less automated. :)

Reply:
Let's keep an eye on `NativeActivity` and script / Android Studio usage, particularly as we build our more Android samples and the Java bindings.

The workflows for

* IREE developers wanting to launch tests across a suite of models and devices from a scripting interface
* IREE users taking a trained model, compiling it, then bundling it into their app as an asset

May seem different in a few ways, but we should aim to share infrastructure between them where possible. We shouldn't expect app developers to fork this script, for example.

Reply:
SGTM!